src/components/pages/user/leaderboard/index.tsx

```diff
@@ -15,33 +15,47 @@ import UserLeaderboardTable from "./table";
 const STOPS = [20, 100, 200, 500];
 
 function UserLeaderboardComponent() {
-  const [limit, setLimit] = useState(STOPS[0]);
-  const [mod, setMod] = useState(LEADERBOARD_MODULES[Object.keys(LEADERBOARD_MODULES)[0]]);
-  const [time, setTime] = useState(
-    Object.keys(LEADERBOARD_STOPS)[Object.keys(LEADERBOARD_STOPS).length - 1]
-  );
+  const { setFilter, filter } = useLeaderboardFilter();
   const { t } = useTranslation();
 
+  const onChangeModule = (v) => {
+    setFilter((_draft) => {
+      _draft.f.module = v.target.value;
+    });
+  };
+
+  const onChangeTime = (v) => {
+    setFilter((_draft) => {
+      _draft.f.period = v.target.value;
+    });
+  };
+
+  const onChangeCount = (v) => {
+    setFilter((_draft) => {
+      _draft.f.limit = +v.target.value;
+    });
+  };
```

Comment:
Instead of using three functions just take the event and extract value and name both from them and this can be just a single function called setFilter? also there's no need to parse limit since GET request will convert it to string anyway

Reply:
also this can directly be handled in hook